R/learner_mgcv_regr_gam.R

```diff
@@ -0,0 +1,176 @@
+        ParamDbl$new("gamma", default = 1, lower = 1, tags = "train"),
+        ParamUty$new("paraPen", default = NULL, tags = "train"),
+        ParamUty$new("G", default = NULL, tags = "train"),
+        ParamUty$new("in.out", default = NULL, tags = "train"),
+        ParamLgl$new("drop.unused.levels", default = TRUE, tags = "train"),
+        ParamLgl$new("drop.intercept", default = FALSE, tags = "train"),
+        ParamInt$new("nthreads", default = 1L, lower = 1L, tags = c("train", "control")),
+        ParamDbl$new("irls.reg", default = 0.0, lower = 0, tags = c("train", "control")),
+        ParamDbl$new("epsilon", default = 1e-07, lower = 0, tags = c("train", "control")),
+        ParamInt$new("maxit", default = 200L, tags = c("train", "control")),
+        ParamLgl$new("trace", default = FALSE, tags = c("train", "control")),
+        ParamDbl$new("mgcv.tol", default = 1e-07, lower = 0, tags = c("train", "control")),
+        ParamInt$new("mgcv.half", default = 15L, lower = 0L, tags = c("train", "control")),
+        ParamDbl$new(
+          "rank.tol",
+          default = .Machine$double.eps^0.5,
+          lower = 0,
+          tags = c("train", "control")
+        ),
+        ParamUty$new("nlm", default = list(), tags = c("train", "control")),
+        ParamUty$new("optim", default = list(), tags = c("train", "control")),
+        ParamUty$new("newton", default = list(), tags = c("train", "control")),
+        ParamInt$new("outerPIsteps", default = 0L, lower = 0L, tags = c("train", "control")),
+        ParamLgl$new("idLinksBases", default = TRUE, tags = c("train", "control")),
+        ParamLgl$new("scalePenalty", default = TRUE, tags = c("train", "control")),
+        ParamInt$new("efs.lspmax", default = 15L, lower = 0L, tags = c("train", "control")),
+        ParamDbl$new("efs.tol", default = .1, lower = 0, tags = c("train", "control")),
+        ParamFct$new(
+          "scale.est",
+          levels = c("fletcher", "pearson", "deviance"),
+          default = "fletcher",
+          tags = c("train", "control")
+        ),
+        ParamLgl$new("edge.correct", default = FALSE, tags = c("train", "control")),
+        ParamInt$new("block.size", default = 1000L, tags = "predict"),
+        ParamLgl$new("unconditional", default = FALSE, tags = "predict")
+      ))
+
+      super$initialize(
+        id = "regr.gam",
+        packages = "mgcv",
+        feature_types = c("logical", "integer", "numeric"),
+        predict_types = c("response", "se"),
+        param_set = ps,
+        properties = c("weights"),
+        man = "mlr3extralearners::mlr_learners_regr.gam"
+      )
+    }
+
+
+  ),
+
+  private = list(
+    .train = function(task) {
+
+      pars = self$param_set$get_values(tags = "train")
+
+      # set column names to ensure consistency in fit and predict
+      self$state$feature_names = task$feature_names
+
+      data = as.data.frame(task$data(cols = c(task$feature_names, task$target_names)))
```

Comment:
`data = as.data.frame(task$data())`

Does it need to be a data.frame though? data.table inherits from data.frame

Reply:
You are right, it seems to work well without. Removed.